### PR TITLE
[Login] Jetpack connection Part 1

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of: #7299 

Requirements before merging:
- [x] Merging https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2522 and updating fluxc version commit hash.
- [x] Merging #7330 

### Description
This PR adds the first part of handling Jetpack connection during login with site credentials, it adds a button to the account mismatch error that allows starting the Jetpack connection in a WebView, and it detects the end of the connection.
The next steps of the connection will be handled in next PRs.

### Testing instructions

1. Prepare a site that has two accounts, where one of them is connected to Jetpack.
2. Open the app, and sign in using the site address.
3. Use site credentials for login, and enter credentials of the non-connected account.
4. Click on "Connect Jetpack to your account"
5. Continue the connection on the WebView.
6. Confirm the WebView is closed after finishing the connection, and the logcat contains a message: "Jetpack Connected"

### Images/gif
https://user-images.githubusercontent.com/1657201/188423223-fd1c84cd-2ce5-4cb0-9729-e2bb29d5d202.mov

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt [hunk 1]
```diff
@@ -195,12 +229,60 @@ private fun ButtonBar(
...
+
+                        private fun decideUrl(url: String) {
+                            val urlWithoutScheme = url.replace("^https?://".toRegex(), "")
+                            if (viewState.successConnectionUrls.any { urlWithoutScheme.startsWith(it) }) {
+                                viewState.onConnected()
+                            }
+                        }
+                    }
+                    this.webChromeClient = object : WebChromeClient() {
+                        @Suppress("MagicNumber")
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt]: Np, I think this is not needed

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchRepository.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.wordpress.android.login.util.SiteUtils
+import javax.inject.Inject
+
+class AccountMismatchRepository @Inject constructor(
+    private val jetpackStore: JetpackStore,
+    private val siteStore: SiteStore
+) {
+    suspend fun getSiteByUrl(url: String): SiteModel? = withContext(Dispatchers.IO) {
+        SiteUtils.getSiteByMatchingUrl(siteStore, url)
+    }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchRepository.kt:17]: Why the need of changing dispatcher here? Or even making it a suspendable function? This is not hitting the network or disk and looks like a simple logic inside `getSiteByMatchingUrl`

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
The code works as expected and looks good! Just left a couple of non-blocking nitpicks. 

Great job @hichamboushaba 🏅

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchErrorScreen.kt]: Nice catch, I removed it in 06e0dda

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchRepository.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchRepository.kt:17]: `getSiteByMatchingUrl` will get the sites list using `SiteStore.sites` property, and this property [queries](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/9f50893044336efd2f318af97458f523b6bc8c7a/fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.kt#L991-L992) them from the DB, so switching the dispatcher is needed.

I think it's a bad practice to make a DB call inside a property's `get`, as it's easy to miss that the call is heavy, but I think fixing the root issue requires a project by itself, and it's not something we can squeeze in a small task like this.

Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/accountmismatch/AccountMismatchRepository.kt:17]: Oh! I missed that. Thanks for clarifying, and yes, I agree hiding DB calls in property's get can be risky.

**Pull request merged**